### ISSUE TITLE
Startup error

### ISSUE THREAD

**Issue by templarundead** [ish-app/ish#625]
I reinstalled the app and get an error on startup

![image](https://user-images.githubusercontent.com/15968502/73150360-0d175280-40d7-11ea-965b-cc37e866bee0.jpg)

**Comment by tbodt** [comment]
Please press "k" and do this: https://github.com/tbodt/ish/wiki/Extracting-a-dump-of-your-files

**Comment by templarundead** [comment]
Latest iOS/iSH betas

**Comment by templarundead** [comment]
I don't have any PC/Mac

**Comment by tbodt** [comment]
There's unfortunately no other way for me to debug. If anyone comes across this issue with the same problem, please do that.

**Comment by templarundead** [comment]
OK

**Comment by templarundead** [comment]
Just updated to iOS 13.3.1 and its work

**Comment by tbodt** [comment]
Someone sent me a files dump and I found something pretty interesting: the alpine filesystem is pretty much pristine, but the sqlite metadata database was missing a ton of paths. The missing paths are all present somewhere in the database file, meaning either they used to be there but were deleted, or there's a bug in sqlite (unlikely).

The only SQL in iSH that can delete from the paths table is in the handler for unlink() (which would actually delete the file and that hasn't happened) and in the fakefs rebuild code. The rebuild code runs on first launch so that seems like a prime suspect. Also everything appears twice in the database which is a sign the rebuild code has run and recreated everything. But I don't know what about it could be wrong:

- It hasn't been touched in over a year, and I only started getting a lot of these reports more recently
- The whole rebuild operation is done in one transaction, so there can't be a problem with crashing halfway through
- All the sqlite return codes are checked and it would crash if there's an error. On second look there are some calls with missing error checks...
